### PR TITLE
MANTA-5362 multithreaded sharkspotter should concurrently scan _id and _idx

### PR DESCRIPTION
**Pull request by rjloura**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -395,13 +395,20 @@ where
 
         remaining = largest_id - start_id + 1;
 
+        // Find the percent value rounded to the thousand-th of a percent.
+        let percent_complete =
```
Comment (bowrocker) [inline src/lib.rs:399]: Just wondering if you might want to check for remaining is > largest_id, leading to negative percent?

File: src/lib.rs [hunk 2]
```diff
@@ -520,6 +527,45 @@ where
...
+                obj_tx
+                    .send(msg)
+                    .map_err(|e| Error::new(ErrorKind::Other, e.description()))
+            },
+        ) {
+            error!(
+                &log,
+                "Encountered error scanning shard {} ({})", shard_num, e
+            );
+            // TODO panic if this is not an `_idx` or `_id` not found.
```
Comment (bowrocker) [inline src/lib.rs:564]: is the TODO handled?

Comment (rjloura) [reply src/lib.rs]: I have a separate jira issue opened for that.  I need to think about the correct solution a bit more, and it different enough that I didn't want to include it in this change.

File: src/lib.rs (hunk 1 above)
Comment (rjloura) [reply src/lib.rs:402]: added some asserts.

**Pull request merged**